sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/documentanalysis/DocumentAnalysisClient.java

```diff
@@ -214,6 +265,122 @@ public final class DocumentAnalysisClient {
     public SyncPoller<OperationResult, AnalyzeResult>
         beginAnalyzeDocument(String modelId, BinaryData document,
                              AnalyzeDocumentOptions analyzeDocumentOptions, Context context) {
-        return client.beginAnalyzeDocument(modelId, document, analyzeDocumentOptions, context).getSyncPoller();
+        Objects.requireNonNull(document, "'document' is required and cannot be null.");
+        if (CoreUtils.isNullOrEmpty(modelId)) {
+            throw logger.logExceptionAsError(new IllegalArgumentException("'modelId' is required and cannot"
+                + " be null or empty"));
+        }
+
+        final AnalyzeDocumentOptions finalAnalyzeDocumentOptions = getAnalyzeDocumentOptions(analyzeDocumentOptions);
+        context = enableSyncRestProxy(getTracingContext(context));
+        Context finalContext = context;
+        return SyncPoller.createPoller(
+            DEFAULT_POLL_INTERVAL,
+            cxt -> new PollResponse<>(LongRunningOperationStatus.NOT_STARTED, analyzeActivationOperation(modelId,
+                finalAnalyzeDocumentOptions.getPages(),
+                finalAnalyzeDocumentOptions.getLocale(),
+                document,
+                null,
+                finalContext).apply(cxt)),
+            pollingOperation(modelId, finalContext),
+            getCancellationIsNotSupported(),
+            fetchingOperation(modelId, finalContext));
+    }
+
+    private Function<PollingContext<OperationResult>, OperationResult> analyzeActivationOperation(
+        String modelId, List<String> pages, String locale, BinaryData document, String documentUrl, Context context) {
+        return (pollingContext) -> {
+            try {
+                return Transforms.toDocumentOperationResult(analyzeDocument(modelId,
+                    pages,
+                    locale,
+                    document,
+                    documentUrl,
+                    context)
+                    .getDeserializedHeaders().getOperationLocation());
+            } catch (RuntimeException e) {
+                throw logger.logExceptionAsError(e);
+            }
+        };
+    }
+
+    private ResponseBase<AnalyzeDocumentHeaders, Void> analyzeDocument(String modelId, List<String> pages, String locale,
+        BinaryData document, String documentUrl, Context context) {
+        try {
+            if (documentUrl == null) {
+                return service.analyzeDocumentWithResponse(modelId,
+                    null,
+                    CoreUtils.isNullOrEmpty(pages) ? null : String.join(",", pages),
+                    locale,
+                    StringIndexType.UTF16CODE_UNIT,
+                    document,
+                    document.getLength(),
+                    context);
+            } else {
+                return service.analyzeDocumentWithResponse(modelId,
+                    CoreUtils.isNullOrEmpty(pages) ? null : String.join(",", pages),
```

Comment:
NIT: it might be better to have a variable since it is used twice in the method implementation.